Cabal/Distribution/Simple/Program/Types.hs

```diff
@@ -74,10 +74,14 @@ data Program = Program {
        -- | A function to do any additional configuration after we have
        -- located the program (and perhaps identified its version). For example
        -- it could add args, or environment vars.
-       programPostConf :: Verbosity -> ConfiguredProgram -> IO ConfiguredProgram
+       programPostConf :: Verbosity -> ConfiguredProgram -> IO ConfiguredProgram,
+       -- | A function that filters any arguments that don't impact the output
+       -- from a commandline. Used to limit the volatility of dependency hashes
+       -- when using new-build.
+       programFilterArgs :: Maybe Version -> [String] -> [String]
```

Comment:
name bikeshedding: I'd rather call this something like `programNormalizeArgs` to emphasize this is about normalizing

Reply:
I don't have a strong preference for the name, but I hate trying to come up with them, so I'm fine with whatever the majority wants.

Reply:
I am with Herbert regarding the name.